sklearn/utils/validation.py

```diff
@@ -919,7 +919,7 @@ def is_sparse(dtype):
         )
         if all(isinstance(dtype_iter, np.dtype) for dtype_iter in dtypes_orig):
             dtype_orig = np.result_type(*dtypes_orig)
-        elif pandas_requires_conversion and any(d == object for d in dtypes_orig):
+        elif pandas_requires_conversion and any(d is object for d in dtypes_orig):
```

Comment:
Apparently here we change the behaviour because of the following rule:

```
In [32]: np.dtype("object") is object
Out[32]: False

In [33]: np.dtype("object") == object
Out[33]: True
```